infra/src/test/kotlin/fi/fta/geoviite/infra/ui/pagemodel/common/E2EList.kt

```diff
@@ -62,10 +69,30 @@ abstract class E2EList<T>(listBy: By, val itemsBy: By) : E2EViewFragment(listBy)
 
     fun select(check: (T) -> Boolean) = select(getItemWhenMatches(check))
 
+    fun unselect(check: (T) -> Boolean) = unselect(getItemWhenMatches(check))
+
+    private fun itemTextContains(container: WebElement, text: String): Boolean {
+        return container.text.contains(text)
+    }
+
     open fun select(item: T): E2EList<T> = apply {
         logger.info("Select item $item")
 
-        getElementWhenMatches { it == item }.first.click()
+         getElementWhenMatches { it == item }.first.let { match ->
+             val selectedItem = selectedItemBy?.let { getElementIfExists(selectedItemBy) }
+             if (selectedItem == null || selectedItem.let { selected -> !itemTextContains(match, selected.text) })
+                 match.click()
+         }
+    }
+
+    open fun unselect(item: T): E2EList<T> = apply {
+        logger.info("Unselect item $item")
+
+        getElementWhenMatches { it == item }.first.let { match ->
+            val selectedItem = selectedItemBy?.let { getElementIfExists(selectedItemBy) }
+            if (selectedItemBy != null && selectedItem?.let { selected -> itemTextContains(match, selected.text) } ?: false)
+                match.click()
+        }
```

Comment:
Oma funktio unselectille, sillä eräs testi halusi tehdä unselectin. Aiemmin siinä oltiin vaan luotettu siihen, että `select`:in tekeminen uudestaan jo selectatulle riville unselectaa sen. Nykyään täytyy olla eksplisiittinen.